_posts/2025-03-05-#31405.md

```diff
@@ -7,20 +7,46 @@ authors: [mzumsande]
 components: ["validation"]
 host: stickies-v
 status: upcoming
-commit:
+commit: 4ba2e480ffa0b77113953bee4ff5c9349e277e7e
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- The [`BlockManager::m_block_index`](https://github.com/bitcoin/bitcoin/blob/3c1f72a36700271c7c1293383549c3be29f28edb/src/node/blockstorage.h#L277) is a map which is used to keep track of which block headers exist, how they interconnect, and where on disk they are stored [^1]. Generally speaking, it is updated whenever a new valid header with sufficient Proof-of-Work (PoW) is received. It contains entries for blocks in the main chain, alternative chains, and even invalid blocks.
+
+- The `CBlockIndex` objects in this map can be considered the nodes in a tree shaped structure with the genesis block at its root. By definition, each block can only point to a [single predecessor](https://github.com/bitcoin/bitcoin/blob/3c1f72a36700271c7c1293383549c3be29f28edb/src/chain.h#L147), but multiple blocks can point to the same predecessor. Of course, in a single chain, this tree is pruned so that each block will never have more than one block pointing to it. The entire tree structure is kept to enable efficiently handling chain reorgs.
+
+- Block validation is a process that consists of multiple steps, gradually bumping up the `CBlockIndex`'s `nStatus` from its default-initialized `BLOCK_VALID_UNKNOWN` to `BLOCK_VALID_SCRIPTS`. The validation flow which is relevant to this PR can be summarised as:
+	- When the header is successfully validated with `AcceptBlockHeader()`, it is added to `m_block_index`, and generally never removed from it anymore, even if the block fails a subsequent validation step.
+	- When the partial block validation in `AcceptBlock()` succeeds, the block is persisted to disk. This validation is partial, because full validation requires the predecessor block's data, which the node may not have received yet.
+	- As soon as a block header becomes part of the fully validated current chain, `ActivateBestChain()` will attempt to connect the block. If this final validation succeeds, the block is connected and its `nStatus` is [raised](https://github.com/bitcoin/bitcoin/blob/3c1f72a36700271c7c1293383549c3be29f28edb/src/validation.cpp#L2762) to the ultimate `BLOCK_VALID_SCRIPTS`.
```

Comment:
nit: this sounds a bit circular, maybe: "As soon as a block header becomes _eligible_ to be part of the fully validated current chain" (most PoW, all predecessors have full data).